### ISSUE TITLE
[Analyze] Validate stack IDs

### ISSUE THREAD

**Issue by natalieparellano** [buildpacks/lifecycle#471]
The buildpacks [spec](https://github.com/buildpacks/spec/blob/main/buildpack.md#process) states that

> The lifecycle MUST run `/bin/detect` for all buildpacks in a group in a container using common stack with a common set of mixins. The lifecycle MUST fail detection if any of those buildpacks does not list that stack in `buildpack.toml`. The lifecycle MUST fail detection if any of those buildpacks specifies a mixin associated with that stack in `buildpack.toml` that is unavailable in the container.

However, the lifecycle does not currently validate stack IDs (leaving this up to the platform). 

The lifecycle should validate that the provided buildpacks and the provided stack can work together by validating that all buildpacks list the provided stack (or `*`, if on a buildpack API that contains https://github.com/buildpacks/rfcs/blob/main/text/0056-any-stack-buildpacks.md). The lifecycle can examine `CNB_STACK_ID` to know what stack was provided.

**Comment by ekcasey** [comment]
In addition, we should check that the following all match
1. The previous image stack ID
2. The build environment stack ID
3. The run image stack ID

We should also consider checking whether the layers in the cache created for the correct stack ID. If we decided to do this:
1. We will need to store stack ID in the cache metadata.
2. For backwards compatibility, in cases where the cache metadata is missing the stack ID we should be optimistic and assume the layers were created for the correct stack.

**Comment by natalieparellano** [comment]
@jabrown85 do we want to move this to 0.12.0? I think it'll go nicely with https://github.com/buildpacks/spec/issues/206

**Comment by jabrown85** [comment]
That sounds good to me @natalieparellano

**Comment by natalieparellano** [comment]
> We should also consider checking whether the layers in the cache were created for the correct stack ID. If we decided to do this:

>We will need to store stack ID in the cache metadata.
For backwards compatibility, in cases where the cache metadata is missing the stack ID we should be optimistic and assume the layers were created for the correct stack.

I think this probably makes sense to be a separate issue.

**Comment by natalieparellano** [comment]
Q: If stack IDs are the same, can we assume that the images have the same platform & architecture?

**Comment by natalieparellano** [comment]
Blocked on https://github.com/buildpacks/lifecycle/issues/530

**Comment by natalieparellano** [comment]
Removing this from 0.12.0 milestone for the same reason as https://github.com/buildpacks/lifecycle/issues/573#issuecomment-867809036
